frontend/benefit/applicant/src/components/applications/uploadAttachment/useUploadAttachment.ts

```diff
@@ -0,0 +1,64 @@
+import { ATTACHMENT_TYPES } from 'benefit/applicant/constants';
+import { showErrorToast } from 'benefit/applicant/utils/common';
+import React from 'react';
+
+type ExtendedComponentProps = {
+  handleUploadClick: () => void;
+  handleUpload: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  uploadRef: React.RefObject<HTMLInputElement>;
+};
+
+const useUploadAttachment = (
+  attachmentType: ATTACHMENT_TYPES,
+  allowedFileTypes: string[],
+  maxSize: number,
+  errorTitle: string,
+  errorFileSizeText: string,
+  errorFileTypeText: string,
+  onUpload: (data: FormData) => void
+): ExtendedComponentProps => {
+  const uploadRef = React.createRef<HTMLInputElement>();
+
+  const resetUploadInput = React.useCallback(() => {
+    if (uploadRef.current?.value) {
+      uploadRef.current.value = '';
+    }
+  }, [uploadRef]);
+
+  const handleUploadClick = (): void => {
+    void uploadRef?.current?.click();
+  };
+
+  const handleUpload = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const file = e.target.files?.[0];
```

Comment:
I would have checked here `if (!file) return;` and continued the rest of the function body knowing that `file` and `fileSize` have truthy values.